ocean_lib/models/test/test_ddo.py

```diff
@@ -37,6 +39,93 @@ def get_ddo_sample(datatoken_address):
     return asset
 
 
+def test_ddo_credentials_addresses_both():
+    """Tests DDO credentials when both deny and allow lists exist on the asset."""
+    sample_ddo_path = get_resource_path("ddo", "ddo_sa_sample_with_credentials.json")
+    assert sample_ddo_path.exists(), "{} does not exist!".format(sample_ddo_path)
+
+    ddo = DDO(json_filename=sample_ddo_path)
+    assert ddo.get_addresses_of_type("allow") == ["0x123", "0x456a"]
+    assert ddo.get_addresses_of_type("deny") == ["0x2222", "0x333"]
+    assert (
+        ddo.get_address_allowed_code({"type": "address", "value": "0x111"})
```

Comment:
Based on my recommendation to remove the `get_address_allowed_code` method, I recommend that you use `is_consumable` here like so:

```python
assert (
    ddo.is_consumable({"type": "address", "value": "0x111"}, with_connectivity_check=False) == ConsumableCodes.CREDENTIAL_NOT_IN_ALLOW_LIST
)
```

Extrapolate this recommendation to all other uses of `get_address_allowed_code`

Reply:
Addressed that in separation of concerns. And, as you can see, tests are also easier to do when functions handle separate concerns. I am interested in testing ONLY address access here, not connectivity or disabling. Using your proposed version of `is_consumable`, I would have to bundle up all these tests together.